tests/unit/test_charm_simulation_action.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import tempfile
+
+import scenario
+
+from tests.unit.fixtures import UEFixtures
+
+
+class TestCharmConfigure(UEFixtures):
+    def test_given_ping_transmits_packets_correctly_when_start_simulation_action_then_action_status_is_successful(  # noqa: E501
+        self,
+    ):
+        test_successful_stdout = "10 packets transmitted, 10 received, 0% packet loss, time 9012ms"
+        with tempfile.TemporaryDirectory() as temp_dir:
+            self.mock_check_output.return_value = b"1.2.3.4"
+            rfsim_relation = scenario.Relation(
+                endpoint="fiveg_rfsim",
+                interface="fiveg_rfsim",
+                remote_app_data={
+                    "rfsim_address": "1.1.1.1",
+                },
+            )
+            config_mount = scenario.Mount(
+                src=temp_dir,
+                location="/tmp/conf",
+            )
+            container = scenario.Container(
+                name="ue",
+                can_connect=True,
+                mounts={
+                    "config": config_mount,
+                },
+                exec_mock={
+                    ("ping", "-I", "oaitun_ue1", "8.8.8.8", "-c", "10"): scenario.ExecOutput(
+                        return_code=0,
+                        stdout=test_successful_stdout,
+                        stderr="",
+                    ),
+                },
+            )
+            state_in = scenario.State(
+                leader=True,
+                relations=[rfsim_relation],
+                containers=[container],
+            )
+
+            action_results = self.ctx.run_action("start-simulation", state_in)
+
+            assert action_results.success
+            assert action_results.results["success"] == "true"  # type: ignore[reportOptionalSubscript]  # noqa: E501
+            assert action_results.results["result"] == test_successful_stdout  # type: ignore[reportOptionalSubscript]  # noqa: E501
```

Comment:
We probably can remove the `type:ignore...` with this before checking the specific keys

```
assert action_results.results
```

Reply:
I wouldn't do it in the charm logic, but in test I don't think it really matters - we don't really care if it fails on a missing key or invalid value for that key. Both situations mean that something went wrong and the test should fail.

Reply:
Well it's easier for a dev to see what the error is if the assertion is precise. Plus it's generally bad practice to ignore static analysis checks, they're there for a reason.

Reply:
Removed